### PR TITLE
Add some new functions to make it easier to monitor draining/down hosts

### PR DESCRIPTION
**Pull request by nhandler**
These new functions will make it easier to add monitoring checks to detect hosts that are forgotten in a draining/down state.

### CODE REVIEW

File: paasta_tools/mesos_maintenance.py [hunk 1]
```diff
@@ -200,6 +200,49 @@ def is_host_down(hostname=getfqdn()):
...
+def get_hosts_forgotten_draining():
+    """Find hosts that are still marked as draining (rather than down) after the start
+    of their maintenance window. For each of these hosts, either mark them as 'down' for
+    maintenance, adjust the time of their maintenance window, or cancel the maintenance
+    window if it is no longer needed.
+    :returns: a list of hostnames of hosts forgotten draining
+    """
+    draining_hosts = get_draining_hosts()
+    hosts_past_maintenance_start = get_hosts_past_maintenance_start()
+    print("draining_hosts: %s" % draining_hosts)
```
Comment (mattmb) [inline paasta_tools/mesos_maintenance.py]: Can we make these `log.something` please

File: paasta_tools/mesos_maintenance.py [hunk 2]
```diff
@@ -200,6 +200,49 @@ def is_host_down(hostname=getfqdn()):
     return hostname in get_down_hosts()
 
 
+def get_hosts_forgotten_draining():
+    """Find hosts that are still marked as draining (rather than down) after the start
+    of their maintenance window. For each of these hosts, either mark them as 'down' for
+    maintenance, adjust the time of their maintenance window, or cancel the maintenance
+    window if it is no longer needed.
+    :returns: a list of hostnames of hosts forgotten draining
```
Comment (Rob-Johnson) [inline paasta_tools/mesos_maintenance.py]: These docs aren't accurate (right?)- can you remove the ``For each of these hosts, either mark them as 'down' for maintenance, adjust the time of their maintenance window, or cancel the maintenance window if it is no longer needed.`` please?

File: paasta_tools/mesos_maintenance.py (hunk 1 above)
Comment (nhandler) [inline paasta_tools/mesos_maintenance.py]: Heh, these prints should not have been included, but I can add some `log.debug` calls.

File: paasta_tools/mesos_maintenance.py (hunk 2 above)
Comment (nhandler) [inline paasta_tools/mesos_maintenance.py]: The docs are accurate, but they would be better suited as a `Tip` on a `monitoring_check` than in the function definition. They describe actions a human operator can take to deal with hosts that are forgotten draining. I'll move them.

**Comment by nhandler** [comment]
@mattmb Most of our maintenance will be managed by tools. I would also like to be able to use the start of the maintenance window as a sort of timeout on draining (hence these functions). In those situations, a forgotten draining host will likely be a sign that some tool is failing to timeout the draining process by transitioning the box to the `down` state or by undraining the box if we destroyed it and it no longer exists.

**Pull request merged**